### PR TITLE
P2P Enable adding deposit account as payment method

### PR DESCRIPTION
**Pull request by marcaaron**
cc @stitesExpensify @nickmurray47 if you guys want to also take a look to make sure this will line up with any active changes for "Send Money" and "Payment Methods".

### Details

Adds the ability to set up a personal bank account via the payments page

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/187883

### Tests / QA Steps
1. Create a new account or remove all payment methods from an existing account before testing (use `expensify.com` site as `new.expensify.com` does not yet allow removing payment methods).
2. Navigate to `Settings > Payments > Add payment method` and select "Bank account"
3. Verify the Plaid modal is presented and login with Chase test credentials.
4. Select an account from the list 
5. Tap "Save & continue" without entering a password.
5. Verify an error message appears
6. Enter incorrect password and verify the error shows again
7. Enter correct password and submit again
8. Verify you are brought back to the "Payments" view and the account is now present in the list.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### iOS

320b4beb64.png)

### CODE REVIEW

**Comment by marcaaron** [comment]
Putting this into reviews while I continue to test on other platforms.

**Comment by marcaaron** [comment]
This is probably not a blocker, but one interesting / weird thing discovered while testing is that on Android (but not any other platform) the plaid payments list shows "Wells Fargo" and the correct icon is not displayed. :lolwut:

**Review by ctkochan22 — APPROVED** [review]
Looks good and straight-forward. Would love to get another review though, still ramping up on App issues

**Comment by marcaaron** [comment]
Sounds good, thanks @ctkochan22. Re-rolling with PullerBear to get another set of eyes. 🙇

File: src/components/AddPaymentMethodMenu.js
Comment (nickmurray47) [inline src/components/AddPaymentMethodMenu.js:19]: nab: prop-type doc

File: src/libs/actions/BankAccounts.js
Comment (stitesExpensify) [inline src/libs/actions/BankAccounts.js:84]: Do we want success growls here?

File: src/libs/actions/BankAccounts.js
Comment (stitesExpensify) [inline src/libs/actions/BankAccounts.js:95]: Failure growl here

**Review by stitesExpensify — APPROVED** [review]
Code LGTM, shouldn't interfere with my projects. Leaving it to @nickmurray47 for final review

**Review by nickmurray47 — APPROVED** [review]
Gave it a look and also looks pretty good to me!

**Review by yuwenmemon — APPROVED** [review]

File: src/libs/actions/BankAccounts.js
Comment (marcaaron) [reply src/libs/actions/BankAccounts.js:84]: Dang yea that would probably be a nice touch. It's pretty strange that we just kick you back to the payments page now that I think about it. But it works for now.

**Pull request merged**
